aws/resource_aws_launch_template.go

```diff
@@ -0,0 +1,976 @@
+					return
+				},
+			},
+
+			"client_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"default_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"latest_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"block_device_mappings": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"no_device": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"virtual_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ebs": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"delete_on_termination": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"encrypted": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"iops": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"kms_key_id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"snapshot_id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"volume_size": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"volume_type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"credit_specification": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cpu_credits": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"disable_api_termination": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"ebs_optimized": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"elastic_gpu_specifications": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"iam_instance_profile": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arn": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"image_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"instance_initiated_shutdown_behavior": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```

Comment:
Let's provide plan time validation for the `instance_initiated_shutdown_behavior` attribute 👍 

```go
ValidateFunc: validation.StringInSlice([]string{
  ec2.ShutdownBehaviorStop,
  ec2.ShutdownBehaviorTerminate,
}, false),
```